### PR TITLE
[incubator-kie-drools#5776] [new-parser] failed to parse when a metho…

### PR DESCRIPTION
**Pull request by tkobayas**
…d name is a drl keyword

**Issue**: 
* https://github.com/apache/incubator-kie-drools/issues/5776

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4 [hunk 1]
```diff
@@ -162,6 +162,76 @@ typeArgument
...
+    | DRL_DATE_EFFECTIVE
+    | DRL_DATE_EXPIRES
+    | DRL_DIALECT
+    | DRL_CALENDARS
+    | DRL_TIMER
+    | DRL_DURATION
+    ;
+
+// matches any identifiers including acceptable java keywords (defined in JavaParser.g4) and drl keywords
+drlIdentifier returns [Token token]
```
Comment (tkobayas) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4:215]: We need to define this rule to aggregate acceptable tokens.

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4 [hunk 2]
```diff
@@ -608,18 +678,18 @@ primary returns [BaseDescr result]
     //|   void_key DOT class_key
     |   inlineMapExpression
     |   inlineListExpression
-    |   i1=IDENTIFIER { helper.emit($i1, DroolsEditorType.IDENTIFIER); }
+    |   i1=drlIdentifier { helper.emit($i1.token, DroolsEditorType.IDENTIFIER); }
```
Comment (tkobayas) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4:681]: I have fixed only `primary` rule to make this PR concise and easy to review. I filed https://github.com/apache/incubator-kie-drools/issues/5784 to review `DRL6Expressions.g4` for other possible `IDENTIFIER` replacement.

**Comment by tkobayas** [comment]
`NullSafeDereferencingTest.testNullSafeDereferencingPredicateMethod` has fixed.

Before PR:
`drools-model-codegen`
```
[ERROR] Tests run: 2433, Failures: 430, Errors: 2, Skipped: 9
```

After PR:
`drools-model-codegen`
```
[ERROR] Tests run: 2433, Failures: 419, Errors: 2, Skipped: 9
```

**Comment by tkobayas** [comment]
@yurloc @gitgabrio @mariofusco Please review, thanks!

**Review by mariofusco — APPROVED** [review]

**Review by yurloc — APPROVED** [review]
LGTM.

**Pull request merged**